contracts/adapters/nativeTokens/OriginAdapter.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.11;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "../../interfaces/IBridge.sol";
+import "../interfaces/IDepositAdapterTarget.sol";
+import "../../interfaces/IBasicFeeHandler.sol";
+
+
+contract OriginAdapter is AccessControl {
+    IBridge public immutable _bridge;
+    bytes32 public immutable _resourceID;
+    IBasicFeeHandler public _feeHandler;
+
+    event Withdrawal(address recipient, uint amount);
+
+    error SenderNotAdmin();
+    error InsufficientMsgValueAmount(uint256 amount);
+    error MsgValueLowerThanFee(uint256 amount);
+    error TokenWithdrawalFailed();
+    error InsufficientBalance();
+
+    modifier onlyAdmin() {
+        if (!hasRole(DEFAULT_ADMIN_ROLE, msg.sender)) revert SenderNotAdmin();
+        _;
+    }
+
+    constructor(IBridge bridge, IBasicFeeHandler feeHandler, bytes32 resourceID) {
+        _bridge = bridge;
+        _resourceID = resourceID;
+        _feeHandler = feeHandler;
```

Comment:
Any reason why the fee handler is set separately instead of being fetched from the bridge when the fee calculation is required?